File: JavaScript/2171. Removing Minimum Number of Magic Beans.js
/**
 * @param {number[]} beans
 * @return {number}
 */
var minimumRemoval = function(beans) {
    var n = beans.length;
    var res = Number.MAX_VALUE;
    var sum = 0;
    beans.sort((a,b) => a-b);
    for(var i of beans){
        sum += i;
    }
    for(var i = 0; i<n; i++){
        res = Math.min(res, sum - (n - i) * beans[i]);
    }
    return res;
};

File: C/2171. Removing Minimum Number of Magic Beans.c
int comp(const void* a, const void* b){
    int int_a = * ( (int*) a );
    int int_b = * ( (int*) b );
    return (int_a > int_b) - (int_a < int_b);
}
long long minimumRemoval(int* beans, int beansSize){
    long long res = LONG_MAX;
    long long sum = 0;
    qsort(beans, beansSize, sizeof(int), comp);
    for(int i = 0; i<beansSize; i++){
        sum += beans[i];
    }
    for(int i = 0; i<beansSize; i++){
        res = fmin(res, sum - (long)(beansSize - i) * beans[i]);
    }
    return res;
}
